### PR TITLE
Accessibility for SVG icons

### PR DESCRIPTION
**Pull request by shaal**
Accomplished by adding the following attributes: role, title & desc

Based on documentation:
https://fontawesome.com/how-to-use/on-the-web/other-topics/accessibility#svg-semantic
https://css-tricks.com/accessible-svgs/
https://www.sitepoint.com/tips-accessible-svg/

(added a [Facebook icon](https://fontawesome.com/icons/facebook-f) from the [FontAwesome library](https://fontawesome.com/icons))

### CODE REVIEW

**Comment by shaal** [comment]
It's a good question. 
The way I was created it was if you don't care about accessibility, provide minimum details (the way Emulsify working now)
and If you want accessibility in icons - add all important parts, such as `role`, `title`, and `desc`.
We could change it into checking each attribute individually whether it exists or not.

As I read more about it, we could also add support for decorative icons, those icons that don't change how a user interacts with the website.
Something like -
```
<svg... {{ icon_decorative ? 'aria-hidden="true" role="img"' : '' }}>
```

File: components/_patterns/01-atoms/04-images/icons/_icon.twig [hunk 1]
```diff
@@ -13,6 +16,14 @@
 {% set icon_url = icon_url|default('/themes/custom/emulsify/dist/img/sprite/') %}
 {{ attach_library('emulsify/sprite') }}
 
-<svg {{ bem(icon_base_class, (icon_modifiers), icon_blockname, (icon_js_class)) }}>
+<svg {{ bem(icon_base_class, (icon_modifiers), icon_blockname, (icon_js_class)) }}
+  {% if icon_role %}
```
Comment (ModulesUnraveled) [inline components/_patterns/01-atoms/04-images/icons/_icon.twig]: I think this should check that ALL variables are included. We don't want to accidentally produce something like this if they only provide the role

`aria-labelledby="title-" aria-describedby="desc-" role="icon-role"`

We could get more granular with the `if` statements, but I'm fine with a "provide it all, or provide nothing` approach, and fallback to `aria-hidden` if they don't provide anything

Comment (shaal) [reply components/_patterns/01-atoms/04-images/icons/_icon.twig]: We could use `set` for each attribute on top and then each attribute_placeholder will either add (nothing) or will add the accurate attribute itself.

**Comment by shaal** [comment]
Added individual check for each attribute, and added the option for decorative icon

**Comment by evanmwillhite** [comment]
I'm just gonna pull this in and we can change those in a later PR so you get credit here. Thanks again @shaal for your work here!

**Pull request merged**